### ISSUE TITLE
Add option for doDispatchPropsDependOnOwnProps

### ISSUE THREAD

**Issue by johnzielke** [reactjs/react-redux#515]
I am using [defaultMemoize](https://github.com/reactjs/reselect/blob/master/src/index.js) from the reselect library to memoize my dispatchToProps. Now the problem is, that the memoized function returned by defaultMemoize does not contain any arguments in its "definition" (And it will probably be the same way with any other generic memoization). This results in the function always having "doDispatchPropsDependOnOwnProps" set to true, even though i don't use any ownProps here. There should be an option to set this to false, even if the arguments length is 0. Or am I missing a different, better aproach here?
Here is some sample code to illustrate my current problem:
```
import {defaultMemoize} from "reselect";


const mapStateToProps = function (state, ownProps) {
 .......
};

const makeMapDispatchToProps = () => defaultMemoize(function (dispatch) {
  return {
    setCost: function (entry, cost) {
     ......
    },
    setReturnDate: function (entry, isReturned) {
      .....
    },
    onDelete: function (entry) {
      dispatch(lenddataactions.deleteLendEntry(entry));
    }
  };
});
export default connect(mapStateToProps, makeMapDispatchToProps)(LendEntryItem);
```

I could (and probably will for now) bypass this by only comparing the equality of the first parameter, but that seems quite hacky...

What are your ideas on this? This option should not be too difficult to add (I could create the PR myself), but if I'm missing another possibility please tell me!

**Comment by markerikson** [comment]
Two questions:

- Why do you need to memoize `mapDispatch`?  If you're not asking for the `ownProps` argument, that _should_ only be called once at component creation time.
- While you certainly _can_ define functions inside of `mapDispatch`, defining them _outside_ has a number of advantages.  You can access the plain function definitions to test them, and you can pass them straight to `connect` for binding to `dispatch`.

**Comment by jimbolla** [comment]
Based on what you've got there, I don't see why you need memoize at all.

You could replace makeMapDispatchToProps with:
```js
const mapDispatchToProps = dispatch => ({
    setCost: function (entry, cost) {
     ......
    },
    setReturnDate: function (entry, isReturned) {
      .....
    },
    onDelete: function (entry) {
      dispatch(lenddataactions.deleteLendEntry(entry));
    }
});
```
This function would only be called once when your component is instantiated.

**Comment by johnzielke** [comment]
You are both correct, I always added the second argument when I wrote my code, because I didn't realize this could have any performance impact... I guess it could be great if we added that to the docs, so people won't stumble over this issue in the first place. Although in my case, I should have spotted that.. I should probably get some sleep :)
Closing this issue. You guys are great, thanks for your help with my (nonexistent) problem!
